.github/workflows/formatter.yml

```diff
@@ -0,0 +1,37 @@
+name: Format
+
+on:
+  pull_request:
+    paths: ['**.cs']
+  push:
+    paths: ['**.cs']
+
+permissions:
+  contents: write
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 7.0.x
+
+      - name: Format scripts
+        run: dotnet tool run dotnet-format --fix-whitespace
```

Comment:
Before running the tool we need to restore it:
```
  - name: Format code
        run: |
          dotnet tool restore
          dotnet tool run dotnet-format --fix-whitespace
```

Reply:
You are right, I originally thought that `dotnet-format` implicitly restores.